slog/slog_test.go

```diff
@@ -437,3 +437,89 @@ func TestContext(t *testing.T) {
 		t.Errorf("failed to retrieve correct entry from context:\n%s", diff)
 	}
 }
+
+// Examples
```

Comment:
As far as I can tell idiomatic approach is to split this off into an `example_test.go` file.